File: mesrapp-backend/mesrapp/src/main/java/com/example/mesrapp/model/IdMatch.java
package com.example.mesrapp.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "id_matches")
public class IdMatch {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private Integer id;
    @Column(nullable = false)
    private String spotiId;    
    @Column(nullable = false)
    private String matchedSpotiId;
    private String matchedMusicId;
    private String matchedArtistId;
}

File: mesrapp-frontend/mesrapp/app/src/main/java/com/example/mesrapp/models/IdMatch.kt
package com.example.mesrapp.models

import androidx.room.ColumnInfo
import androidx.room.Entity

@Entity
data class IdMatch (
    @ColumnInfo(name = "id", index = true) val id: String,
    @ColumnInfo(name = "spotiId") val spotiId: String,
    @ColumnInfo(name = "matchedSpotiId") val matchedSpotiId: String = "",
    @ColumnInfo(name = "matchedMusicId") val matchedMusicId: String = "",
    @ColumnInfo(name = "matchedArtistId") val matchedArtistId: String = "",
) {
    constructor(spotiId: String) : this(spotiId, "", "", "", "")
}
